varats/data/provider/cve/cve_map.py

```diff
@@ -0,0 +1,258 @@
+"""
+Map commits with resolved CVE's and CWE's based on multiple strategies.
+
+Example Calls:
+    generate_security_commit_map(
+        path=Path('/home/vara/repos/vim'),
+        vendor='vim',
+        product='vim'
+    )
+    generate_security_commit_map(
+        path=Path('/home/vara/repos/tensorflow'),
+        vendor='google',
+        product='tensorflow'
+    )
+Example Output:
+    {
+        799: {
+            'commit': '76b92b2830841fd4e05006cc3cad1d8f0bc8101b',
+            'cve': [CVE-2008-3432],
+            'cwe': []
+        },
+        [..]
+    }
+"""
+import logging
+import typing as tp
+from collections import defaultdict
+from pathlib import Path
+import re
+from packaging.version import parse as parse_version, Version, LegacyVersion
+from plumbum import local
+from plumbum.cmd import git
+from varats.data.provider.cve.cve import (CVE, CWE_LIST, find_cve, find_cwe,
+                                          find_all_cve, CWE)
+
+LOG = logging.getLogger(__name__)
+
+
+def __n_grams(text: str,
+              filter_reg: str = r'[^a-zA-Z0-9]',
+              filter_len: int = 3) -> tp.Set[str]:
+    """
+    Divide some text into character-level n-grams.
+
+    Args:
+        text: the text to compute the n-grams for
+        filter_reg: regular expression to filter out special characters
+                    (default: non-letters)
+
+    Return:
+        the set of n-grams
+    """
+    results: tp.Set[str] = set()
+    filter_comp: tp.Pattern[str] = re.compile(filter_reg)
+    for word in text.split():
+        if filter_reg:
+            word = filter_comp.sub('', word)
+        word = word.strip()
+        if filter_len <= len(word):
+            results.add(word)
+    return results
+
+
+def __collect_via_commit_mgs(
+    commits: tp.List[str]
+) -> tp.Dict[str, tp.Dict[str, tp.Set[tp.Union[CVE, CWE]]]]:
+    """
+    Collect data about resolved CVE's/CWE's using the commit message.
+
+    Args:
+        commits: a list of commits
```

Comment:
```suggestion
        commits: a list of commits in textual form
```
as I understand it, commit means the full commit as a text?
should we also rename the arg name?